src/renderer/components/wizard/wizard.tsx

```diff
@@ -190,7 +190,9 @@ export class WizardStep<D> extends React.Component<WizardStepProps<D>, WizardSte
     if (this.form.noValidate || this.form.checkValidity()) {
       this.next();
     }
-  }, 100);
+  }, 100, {
+    leading: true,
```

Comment:
Can you explain what it does?

Reply:
Sure.

This configures how the `debounce`-ed function is called when `this.submit` is called. Namely, it now calls the inner function immediately and then won't call it again for at least another 100ms. Whereas it used to wait for the first 100ms before calling the inner function.

This article (at the given heading and the next heading of "leading edge") explains it very well.

https://css-tricks.com/debouncing-throttling-explained-examples/#aa-debounce 

---

As to why I needed to do this:

It was for getting the unit tests to pass. If the inner function is only called after 100ms then I would have to add that amount of delay into the test (which cases other problems), but with this configuration change the action happens immediately (and subsequent calls within 100ms are then just ignored) makes the tests easier to write and understand.

Reply:
> It was for getting the unit tests to pass. 

Test shouldn't require this. In case you want to make tests pass without this, then you would have to use `useFakeTime` and `advanceFakeTime`. Here, you are changing behaviour to make test pass, which is wrong.

Reply:
It is not wrong if the behaviour was incorrect. The point of this denouncing is to make sure that we don't accidentally double submit.